flexget/plugins/sites/passthepopcorn.py

```diff
@@ -275,6 +303,15 @@ def search(self, task, entry, config):
                     e['golden_popcorn'] = torrent['GoldenPopcorn']
                     e['checked'] = torrent['Checked']
                     e['uploaded_at'] = dateutil_parse(torrent['UploadTime'])
+                    e['imdb_id'] = 'tt%s' % str(movie['ImdbId']).zfill(7)
+
+                    resolution = self._lookup_quality(torrent, 'Resolution', RESOLUTION_MAP)
+                    source = self._lookup_quality(torrent, 'Source', SOURCE_MAP)
+                    codec = self._lookup_quality(torrent, 'Codec', CODEC_MAP)
+                    quality = ' '.join([resolution, source, codec]).strip()
+
+                    if quality:
+                        e['quality'] = quality
```

Comment:
Plugins are not really supposed to set the `quality` field themselves. That's the parser's job.

Reply:
I'm not sure this is really a problem. If a plugin can provide a better quality than the parser could produce I think they should. On the other hand, it should be a `Quality` instance, not a string in that case.